x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/audit/logfile/LoggingAuditTrail.java

```diff
@@ -1627,6 +1631,20 @@ LogEntryBuilder withAuthentication(Authentication authentication) {
                     logEntry.with(PRINCIPAL_REALM_FIELD_NAME, creatorRealmName);
                     // No domain information is needed here since API key itself does not work across realms
                 }
+                if (authentication.isRemoteAccess()) {
+                    final Authentication remoteAuthentication = (Authentication) authentication.getAuthenticatingSubject()
+                        .getMetadata()
+                        .get(AuthenticationField.REMOTE_ACCESS_AUTHENTICATION_KEY);
+                    final StringMapMessage remoteAccessLogEntry = logEntry.newInstance(Collections.emptyMap());
+                    addAuthenticationFieldsToLogEntry(remoteAccessLogEntry, remoteAuthentication, logger);
+                    try {
+                        final XContentBuilder builder = JsonXContent.contentBuilder().humanReadable(true);
+                        builder.map(remoteAccessLogEntry.getData());
+                        logEntry.with(REMOTE_ACCESS_FIELD_NAME, Strings.toString(builder));
+                    } catch (IOException e) {
+                        logger.error("Failed to write remote authentication [{}] as audit log entry", remoteAuthentication);
```

Comment:
The current behaviour for audit logging is that the whole operation would fail if audit logging fails (here is an [example](https://github.com/elastic/elasticsearch/blob/c807a096c02d031cad61d037dc5976af8d2efc8a/x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/audit/logfile/LoggingAuditTrail.java#L773)). This means we should throw error here.

I am not entirely convinced that audit loging should have this dramatic effect. But that's the existing behaviour and I feel it is better to be consistent and have a separate conversation if we want to change this behaviour.

Reply:
Good catch! I intuitively went with not failing the request if audit fails, however, I fully agree on following convention here and possibly revisiting it in a broader conversation.